staging-service/src/main/java/com/github/cafdataprocessing/services/staging/dao/filesystem/FileSystemDao.java

```diff
@@ -234,4 +244,91 @@ private void completeInProgressBatch(final TenantId tenantId, final Path inProgr
         LOGGER.debug("Batch {} completed successfully.", batchId);
     }
 
+    /**
+     *
+     */
+    @Override
+    public void cleanUpStaleInprogressBatches()
+    {
+        if (skipBatchFileCleanup) {
+            return;
+        }
+        try {
+            final List<Path> batchesToClean = Files.list(Paths.get(basePath))
+                .map(p -> getTenantInprogressDirectorySafely(p.getFileName().toString()))
+                .flatMap(p -> getAllBatchFilesForAllDirectories(p.get()))
+                .filter(p -> BatchNameProvider.validateFileName(p.getFileName().toString()))
+                .filter(p -> shouldDelete(p.getFileName().toString()))
+                .filter(p -> checkAllSubfilesSafely(p))
+                .collect(Collectors.toList());
+            for (final Path filepath : batchesToClean) {
+                try {
+                    FileUtils.deleteDirectory(filepath.toFile());
+                } catch (final IOException | IllegalArgumentException ex) {
+                    LOGGER.error("Unable to delete directory {}", filepath);
+                    LOGGER.debug("An error occured while attempting to delete folder {}", filepath, ex);
+                }
+            }
+        } catch (final IOException ex) {
+            LOGGER.error("An exception occured trying to read the files in the base directory.", ex);
+        }
+    }
+
+    private boolean checkAllSubfilesSafely(final Path path)
+    {
+        try {
+            return Files.list(path)
```

Comment:
* <p> The returned stream encapsulates one or more {@link DirectoryStream}s.
     * If timely disposal of file system resources is required, the
     * {@code try}-with-resources construct should be used to ensure that the

Reply:
If I do that wont the steam be closed once this method exits? I want the stream returned to the caller to operate on.